x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/utils.test.ts

```diff
@@ -0,0 +1,658 @@
+  debug: jest.fn((logString: string) => {
+    return logString;
+  }),
+  error: jest.fn((logString: string) => {
+    return logString;
+  }),
+  fatal: jest.fn(),
+  log: jest.fn(),
+};
+describe('utils', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  describe('buildBulkBody', () => {
+    test('if bulk body builds well-defined body', () => {
+      const sampleParams: SignalAlertParams = sampleSignalAlertParams(undefined);
+      const fakeSignalSourceHit = buildBulkBody(sampleDocNoSortId, sampleParams);
+      expect(fakeSignalSourceHit).toEqual({
+        someKey: 'someValue',
+        '@timestamp': 'someTimeStamp',
+        signal: {
+          '@timestamp': fakeSignalSourceHit.signal['@timestamp'], // timestamp generated in the body
+          rule_revision: 1,
+          rule_id: sampleParams.id,
+          rule_type: sampleParams.type,
+          parent: {
+            id: sampleDocNoSortId._id,
+            type: 'event',
+            index: sampleDocNoSortId._index,
+            depth: 1,
+          },
+          name: sampleParams.name,
+          severity: sampleParams.severity,
+          description: sampleParams.description,
+          original_time: sampleDocNoSortId._source['@timestamp'],
+          index_patterns: sampleParams.index,
+          references: sampleParams.references,
+        },
+      });
+    });
+  });
+  describe('singleBulkIndex', () => {
+    test('create successful bulk index', async () => {
+      const sampleParams = sampleSignalAlertParams(undefined);
+      const sampleSearchResult = sampleDocSearchResultsNoSortId;
+      const savedObjectsClient = savedObjectsClientMock.create();
+      const bulkBody = sampleDocSearchResultsNoSortId.hits.hits.flatMap(doc => [
+        {
+          index: {
+            _index: process.env.SIGNALS_INDEX || '.siem-signals-10-01-2019',
+            _id: doc._id,
+          },
+        },
+        buildBulkBody(doc, sampleParams),
+      ]);
+      const mockService: AlertServices = {
+        callCluster: async (action: string, params: BulkIndexDocumentsParams) => {
+          expect(action).toEqual('bulk');
+          expect(params.index).toEqual('.siem-signals-10-01-2019');
+
+          // timestamps are annoying...
+          (bulkBody[1] as SignalHit).signal['@timestamp'] = params.body[1].signal['@timestamp'];
+          expect(params.body).toEqual(bulkBody);
+          return {
+            took: 100,
+            errors: false,
+            items: [
+              {
+                fakeItemValue: 'fakeItemKey',
+              },
+            ],
+          };
+        },
+        alertInstanceFactory: jest.fn(),
+        savedObjectsClient,
+      };
+      const successfulSingleBulkIndex = await singleBulkIndex(
+        sampleSearchResult,
+        sampleParams,
+        mockService,
+        mockLogger
+      );
+      expect(mockLogger.debug).toHaveBeenCalledTimes(2);
+      expect(mockLogger.warn).toHaveBeenCalledTimes(0);
```

Comment:
In general I would be careful about tests that are checking how many times a mock logger debug or warn statement is being called. That gets very specific and a maintainer will be updating these numbers as they either add or remove debug and warn statements from our loggers?

If it's 3 vs 2 times vs 5, etc... I don't know what extra value that provides for the test.

I understand checking if something is called for part of branching logic, but writing a lot of tests around how many times a particular logger debug or warn is being called is going to add more overhead to maintaining and not really provide much value IMHO.

I wouldn't have expects for these